src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -138,6 +138,24 @@ class IndexCollectionManager(
     }
   }
 
+  override def getIndex(indexName: String, logVersion: Int): Option[IndexLogEntry] = {
+    withLogManager(indexName) { logManager =>
+      logManager.getLog(logVersion).map(_.asInstanceOf[IndexLogEntry])
+    }
+  }
+
+  override def getAvailableIndexVersions(indexName: String): Seq[Int] = {
+    withLogManager(indexName) { logManager =>
+      logManager.getActiveIndexVersions().filter { id =>
+        val log = logManager.getLog(id).get
+        val indexDataPaths = log.asInstanceOf[IndexLogEntry].indexContentDirectoryPaths
+        val hadoopConf = spark.sessionState.newHadoopConf()
+        val fs = fileSystemFactory.create(new Path(indexDataPaths.head), hadoopConf)
+        indexDataPaths.forall(path => fs.exists(new Path(path)))
```

Comment:
This is to handle the case where index is vacuumed? How about we move the check to `getActiveIndexVersions` where you stop iterating if you see vacuum operation?

Reply:
It's for vacuumed or manually removed directories (by storage services).